src/main/java/net/mcreator/workspace/references/ReferencesFinder.java

```diff
@@ -0,0 +1,229 @@
+			if (matchFields(me.getGeneratableElement(), Sound.class, (a, t) -> t, false).stream()
+					.anyMatch(e -> e.getUnmappedValue().replaceFirst("CUSTOM:", "").equals(sound.getName())))
+				elements.add(me);
+		}
+
+		return elements;
+	}
+
+	public static List<ModElement> searchStructureUsages(Workspace workspace, String structure) {
+		List<ModElement> elements = new ArrayList<>();
+
+		for (ModElement me : workspace.getModElements()) {
+			if (matchFields(me.getGeneratableElement(), String.class,
+					(a, t) -> a.isAnnotationPresent(StructureReference.class) ? t : null, false).contains(structure))
+				elements.add(me);
+		}
+
+		return elements;
+	}
+
+	public static List<ModElement> searchGlobalVariableUsages(Workspace workspace, String variableName) {
+		List<ModElement> elements = new ArrayList<>();
+
+		for (ModElement me : workspace.getModElements()) {
+			if (!matchFields(me.getGeneratableElement(), String.class, (a, t) -> {
+				if (a.isAnnotationPresent(BlocklyXML.class) && t.contains(
+						"<field name=\"VAR\">global:" + variableName + "</field>"))
+					return t;
+				return null;
+			}, false).isEmpty()) {
+				elements.add(me);
+			}
+		}
+
+		return elements;
+	}
+
+	public static List<ModElement> searchLocalizationKeyUsages(Workspace workspace, String localizationKey) {
+		List<ModElement> elements = new ArrayList<>();
+
+		for (ModElement me : workspace.getModElements()) {
+			GeneratableElement ge = me.getGeneratableElement();
+			if (ge != null && workspace.getGenerator().getElementLocalizationKeys(ge).contains(localizationKey)) {
+				elements.add(me);
+			} else if (!matchFields(ge, String.class, (a, t) -> {
+				if (a.isAnnotationPresent(BlocklyXML.class) && t.contains(localizationKey))
+					return t;
+				return null;
+			}, false).isEmpty()) {
+				elements.add(me);
+			}
+		}
+
+		return elements;
+	}
+
+	private static <T> List<T> matchFields(@Nullable Object source, Class<T> clazz,
+			BiFunction<AccessibleObject, T, T> mapper, boolean methods) {
+		List<T> retVal = new ArrayList<>();
+		if (source == null)
+			return retVal;
+
+		for (Field field : source.getClass().getFields()) {
+			if (!Modifier.isStatic(field.getModifiers())) {
+				try {
+					field.setAccessible(true);
+					checkValue(retVal, field.get(source), field, clazz, mapper, methods);
+				} catch (IllegalAccessException | IllegalArgumentException ignored) {
+				}
+			}
+		}
+		if (methods) {
+			for (Method method : source.getClass().getMethods()) {
+				if (!Modifier.isStatic(method.getModifiers())) {
+					try {
+						method.setAccessible(true);
+						checkValue(retVal, method.invoke(source), method, clazz, mapper, true);
+					} catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException ignored) {
+					}
+				}
+			}
+		}
+
+		return retVal;
+	}
+
+	@SuppressWarnings("unchecked")
+	private static <T> void checkValue(List<T> valuesList, Object value, AccessibleObject field, Class<T> clazz,

```

Comment:
Only fields should be checked, not methods. Fields hold all needed references. It is also unsafe to call methods in some cases, we can't predict what will happen in all cases

Reply:
I block all those dangerous cases by checking the return value type first.

Reply:
And without methods model usages search won't run.

Reply:
Some plugin could add some method that should not be called. You should limit this to methods annotated with some marker